unison-cli/src/Unison/Codebase/Editor/Propagate.hs

```diff
@@ -623,37 +624,25 @@ applyPropagate patch Edits {..} = do
   -- typePreservingTermEdits Patch {..} = Patch termEdits mempty
   --   where termEdits = R.filterRan TermEdit.isTypePreserving _termEdits
 
--- (d, f) when d is "dirty" (needs update),
---             f is in the frontier (an edited dependency of d),
---         and d depends on f
--- a ⋖ b = a depends directly on b
--- dirty(d) ∧ frontier(f) <=> not(edited(d)) ∧ edited(f) ∧ d ⋖ f
+-- | Compute the set of "dirty" references. They each:
 --
--- The range of this relation is the frontier, and the domain is
--- the set of dirty references.
-computeFrontier
+--   1. Depend directly on some reference that was edited in the given patch
+--   2. Have a name in the current namespace (the given Names)
+--   3. Are not themselves edited in the given patch.
+computeDirty
   :: forall m
    . Monad m
   => (Reference -> m (Set Reference)) -- eg Codebase.dependents codebase
   -> Patch
   -> Names
-  -> m (R.Relation Reference Reference)
-computeFrontier getDependents patch names = do
-      -- (r,r2) ∈ dependsOn if r depends on r2
-  dependsOn <- foldM addDependents R.empty edited
-  -- Dirty is everything that `dependsOn` Frontier, minus already edited defns
-  pure $ R.filterDom (not . flip Set.member edited) dependsOn
- where
+  -> m (Set Reference)
+computeDirty getDependents patch names =
```

Comment:
This function is supposed to compute dirty references in the same way as the old version, it just doesn't bother maintaining the `Relation Reference Reference` ("dirty" and "frontier"), since just the `Set Reference` domain ("dirty") is needed by the caller.